codalab/worker/worker.py

```diff
@@ -759,6 +772,23 @@ def bundle_service_reply(self, socket_id, err, message, data):
         else:
             self.bundle_service.reply(self.id, socket_id, message)
 
+    def save_time_stats(self, uuid, from_stage):
+        self.end_stage(uuid, from_stage.stage)
+        self.start_stage(uuid, self.runs[uuid].stage)
+
+    def start_stage(self, uuid, stage):
```

Comment:
If these methods are only going to measure stats, then perhaps the name should reflect that, since `start_stage` in the context of the worker could mean a lot of things.